src/main/webapp/main/main_index.js

```diff
@@ -1,12 +1,21 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
+import { Route, Switch } from "react-router-dom";
 import '../index.css';
 import ErrorHandler from '../components/ErrorHandler';
 import Main from "./Main";
+import RolesManagement from "../rolesManagement/RolesManagement";
+import { BrowserRouter } from 'react-router-dom';
 
 ReactDOM.render(
   <ErrorHandler>
-    <Main/>
+    <BrowserRouter basename= {component.contextPath}>
+      <Switch>
+        <Route path= {"/user/rolesManagement"} render = {(routeProps) =>  <RolesManagement {...routeProps}/> }/>
+        <Route path= {"/project/main"} render = {(routeProps) => <Main {...routeProps}/> }/>
+        <Route path= {"/newConsentGroup/main"} render = {(routeProps) => <Main {...routeProps}/> }/>
```

Comment:
Any props passsed into `RolesManagement` or `Main` will be lost here, they'll only have access to `routeProps`. It should be something like this so that `routeProps` are merged with component `props`:
```suggestion
        <Route path= {"/newConsentGroup/main"} render = {(routeProps) => <Main {...routeProps} {...props} /> }/>
```
I think you also want to pass in props somewhere higher up, maybe in `BrowserRouter` ?

Reply:
@rushtong, please check if my approach makes sense to you. This way it will pass props to each component that is present in Switch.